### ISSUE TITLE
Suggestion: Native depth X searches

### ISSUE THREAD

**Issue by CasperWW** [thinkaurelius/titan#1037]
I would love to see native support for breadth first searches in depth.
Ideally some custom call like:

```
long vertexId = 5965423;
int depth = 4;
String eLabel = "myRelation"
graph.getVertex(vertexId).getVertices(Direction.BOTH, eLabel, depth);
```

Right now I'm doing depth X searches via the [Gremlin Pipe API](https://github.com/tinkerpop/gremlin/wiki/Using-Gremlin-through-Java), but when I compare the performance of depth 2 with a native call (MultiVertexQuery), the native is just so much faster. My guess is that it is because of the Gremlin implementation.
Depth searches would allow for complex queries easy visualisation of the stored graph:
![visualisation of the stored graph](http://s23.postimg.org/7f7304r2z/example.png)

**Comment by mbroecheler** [comment]
@CasperWW, this will be solved with Titan 1.0 which provides multi-query support through Gremlin. Please check out Titan 1.0 once we can release it and let us know whether this meets your demands.

**Comment by mbroecheler** [comment]
Solved in 1.0. Closing for now.
